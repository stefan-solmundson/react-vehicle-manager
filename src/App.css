/*.btnMargin {*/
/*  margin: 0.375rem;*/
/*}*/

.bgDark {
  background-color: rgb(52, 58, 64);
  /*background-color: rgb(42, 50, 56);*/
}

.tt{
  background-color: rgba(170, 102, 204, 0.1);
}

.th:hover {
  background-color: rgba(0, 0, 0, 0.1);
}