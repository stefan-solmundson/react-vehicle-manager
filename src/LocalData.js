/**
 * The local data that the Firebase data originated from
 * @type {{journeysHeadings: *[], journeys: *[], servicesHeadings: *[], vehicles: *[], bookingsDefaultRecord: {bookingStartDate: string, dateCreated: string, dateLastUpdated: string, bookingType: string, bookingEndDate: string, VIN: string, vehicleID: string, bookingID: string, odometerAtStart: number}, refuelsHeadings: *[], services: *[], journeysDefaultRecord: {journeyEndDateTime: string, dateCreated: string, dateLastUpdated: string, journeyStartDateTime: string, VIN: string, vehicleID: string, odometerAtEnd: number, journeyID: string, journeyEndLocation: string, bookingID: string, journeyStartLocation: string, odometerAtStart: number}, refuelsDefaultRecord: {dateCreated: string, dateLastUpdated: string, fuelPrice: number, fuelQuantity: number, VIN: string, vehicleID: string, refuelID: string, bookingID: string}, _pages: [string, string, string, string, string], vehiclesHeadings: *[], refuels: *[], vehiclesDefaultRecord: {odometer: number, fuelCapacity: number, kilometersPerService: number, dateManufactured: number, weight: number, manufacturer: string, dateLastServiced: number, fuelEconomy: number, registrationNumber: string, datePurchased: string, VIN: string, model: string, vehicleID: string, requiresServicing: string}, servicesDefaultRecord: {date: string, serviceLocation: string, odometer: number, VIN: string, vehicleID: string, serviceID: string}, bookings: *[], bookingsHeadings: *[]}}
 */
export const localData = {
  pages: [ "vehicles", "services", "bookings", "journeys", "refuels" ],

  vehicles: [
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      manufacturer: "Jeep",
      model: "Wrangler (JK)",
      odometer: 142000,
      requiresServicing: "Yes",
      dateManufactured: 2017,
      datePurchased: "2017/05/20",
      dateLastServiced: -1,
      // kilometersSinceServiced: 2800000,
      kilometersPerService: 90000,
      fuelEconomy: -1,
      fuelCapacity: 104,
      weight: 1.45,
      registrationNumber: "1ABC-123",
    },
    {
      vehicleID: "Carol's Corolla",
      VIN: "ExampleVIN",
      manufacturer: "Toyota",
      model: "Corolla (E160)",
      odometer: 20000,
      requiresServicing: "No",
      dateManufactured: 2014,
      datePurchased: "2018/05/10",
      dateLastServiced: -1,
      // kilometersSinceServiced: 2000,
      kilometersPerService: 110000,
      fuelEconomy: -1,
      fuelCapacity: 82.5,
      weight: 1.08,
      registrationNumber: "1ABC-123",
    },
    {
      vehicleID: "Carol's 2nd Corolla",
      VIN: "ExampleVIN",
      manufacturer: "Toyota",
      model: "Corolla (E160)",
      odometer: 20000,
      requiresServicing: "No",
      dateManufactured: 2014,
      datePurchased: "2018/06/04",
      dateLastServiced: -1,
      // kilometersSinceServiced: 2000,
      kilometersPerService: 110000,
      fuelEconomy: -1,
      fuelCapacity: 82.5,
      weight: 1.08,
      registrationNumber: "1ABC-123",
    }
  ],
  vehiclesHeadings: [
    {
      label: "Vehicle ID",
      field: "vehicleID",
    },
    {
      label: "VIN",
      field: "VIN",
    },
    {
      label: "Model",
      field: `model`,
    },
    {
      label: "Make",
      field: "manufacturer",
    },
    {
      label: "Odometer (km)",
      field: "odometer",
    },
    {
      label: "Requires Servicing",
      field: "requiresServicing",
    },
    {
      label: "Date Manufactured",
      field: "dateManufactured",
    },
    {
      label: "Date Purchased",
      field: "datePurchased",
    },
    {
      label: "Last Service",
      field: "dateLastServiced",
    },
    // {
    //   label: "kms since Last Service",
    //   field: "kilometersSinceServiced",
    // },
    {
      label: "kms per Service",
      field: "kilometersPerService",
    },
    {
      label: "Fuel Economy",
      field: "fuelEconomy",
    },
    {
      label: "Fuel Capacity (L)",
      field: "fuelCapacity",
    },
    {
      label: "Weight (T)",
      field: "weight",
    },
    {
      label: "Reg. No.",
      field: "registrationNumber",
    },
  ],
  vehiclesDefaultRecord: {
    vehicleID: "Jimbo's Jeep",
    VIN: "ExampleVIN",
    manufacturer: "Jeep",
    model: "Wrangler (JK)",
    odometer: 280000,
    requiresServicing: "Yes",
    dateManufactured: 2017,
    datePurchased: "2017/05/20",
    dateLastServiced: -1,
    // kilometersSinceServiced: 2800000,
    kilometersPerService: 90000,
    fuelEconomy: -1,
    fuelCapacity: 104,
    weight: 1.45,
    registrationNumber: "1ABC-123",
  },

  services: [
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      serviceID: "001-IJ454",
      odometer: 142000,
      date: "2019/04/03",
      serviceLocation: "Autobahn Mechanical and Electrical Services Canningvale",
    },
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      serviceID: "002-FE654",
      odometer: 74000,
      date: "2015/06/01",
      serviceLocation: "Autobahn Mechanical and Electrical Services Victoria Park",
    },
    {
      vehicleID: "Carol's Corolla",
      VIN: "ExampleVIN",
      serviceID: "001-AA000",
      odometer: 14000,
      date: "2018/08/22",
      serviceLocation: "Autobahn Mechanical and Electrical Services Canningvale",
    },
    {
      vehicleID: "Carol's 2nd Corolla",
      VIN: "ExampleVIN",
      serviceID: "001-AA000",
      odometer: 1300,
      date: "2018/08/23",
      serviceLocation: "Autobahn Mechanical and Electrical Services Canningvale",
    },
  ],
  servicesHeadings: [
    {
      label: "Vehicle ID",
      field: "vehicleID",
    },
    {
      label: "VIN",
      field: "VIN",
    },
    {
      label: "Service ID",
      field: "serviceID",
    },
    {
      label: "Odometer Reading",
      field: "odometer",
    },
    {
      label: "Date Last Serviced",
      field: "date",
    },
    {
      label: "Location of Last Service",
      field: "serviceLocation",
    },
  ],
  servicesDefaultRecord: {
    vehicleID: "Jimbo's Jeep",
    VIN: "ExampleVIN",
    serviceID: "051-IJ454",
    odometer: 142000,
    date: "2019/04/03",
    serviceLocation: "Autobahn Mechanical and Electrical Services Canningvale",
  },

  bookings: [
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      bookingID: "002-EE037",
      bookingStartDate: "2019/04/06",
      bookingEndDate: "2019/04/09",
      odometerAtStart: 74000,
      bookingType: "Per Day",
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      bookingID: "001-KD037",
      bookingStartDate: "2019/02/06",
      bookingEndDate: "2019/03/09",
      odometerAtStart: 62333,
      bookingType: "Per Day",
      dateCreated: "2019/02/06",
      dateLastUpdated: "2019/03/09",
    },
    {
      vehicleID: "Carol's Corolla",
      VIN: "ExampleVIN",
      bookingID: "001-II030",
      bookingStartDate: "2019/05/06",
      bookingEndDate: "2019/05/11",
      odometerAtStart: 91000,
      bookingType: "Per km",
      dateCreated: "2019/05/06",
      dateLastUpdated: "2019/05/06",
    },
    {
      vehicleID: "Carol's 2nd Corolla",
      VIN: "ExampleVIN",
      bookingID: "001-KJ022",
      bookingStartDate: "2019/05/06",
      bookingEndDate: "2019/05/11",
      odometerAtStart: 91000,
      bookingType: "Per Day",
      dateCreated: "2019/05/06",
      dateLastUpdated: "2019/05/06",
    },
  ],
  bookingsHeadings: [
    {
      label: "Vehicle ID",
      field: "vehicleID",
    },
    {
      label: "VIN",
      field: "VIN",
    },
    {
      label: "Booking ID",
      field: "bookingID",
    },
    {
      label: "Last Booking Start Date",
      field: "bookingStartDate",
    },
    {
      label: "Last Booking End Date",
      field: "bookingEndDate",
    },
    {
      label: "Odometer When Last Booked (km)",
      field: "odometerAtStart",
    },
    {
      label: "Type",
      field: "bookingType",
    },
    {
      label: "Date Last Booking was Created",
      field: "dateCreated",
    },
    {
      label: "Date Last Booking was Updated",
      field: "dateLastUpdated",
    },
  ],
  bookingsDefaultRecord: {
    vehicleID: "Jimbo's Jeep",
    VIN: "ExampleVIN",
    bookingID: "055-EE037",
    bookingStartDate: "2019/04/06",
    bookingEndDate: "2019/04/09",
    odometerAtStart: 74000,
    bookingType: "Per Day",
    dateCreated: "2019/04/06",
    dateLastUpdated: "2019/04/06",
  },

  journeys: [
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      bookingID: "002-EE037",
      journeyID: "0003-XXI7",
      journeyStartDateTime: "2019/04/06 08:22",
      journeyEndDateTime: "2019/04/09 08:47",
      odometerAtStart: 74018,
      odometerAtEnd: 74033,
      journeyStartLocation: "89080-789777",
      journeyEndLocation: "89080-789889",
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      bookingID: "002-EE037",
      journeyID: "0002-XXI7",
      journeyStartDateTime: "2019/04/06 08:22",
      journeyEndDateTime: "2019/04/09 08:47",
      odometerAtStart: 74000,
      odometerAtEnd: 74018,
      journeyStartLocation: "89080-789777",
      journeyEndLocation: "89080-789889",
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
    {
      vehicleID: "Carol's Corolla",
      VIN: "ExampleVIN",
      bookingID: "002-EE037",
      journeyID: "0002-XXI7",
      journeyStartDateTime: "2019/04/06 08:22",
      journeyEndDateTime: "2019/04/09 08:47",
      odometerAtStart: 74000,
      odometerAtEnd: 74018, journeyStartLocation: "89080-789777",
      journeyEndLocation: "89080-789889",
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
    {
      vehicleID: "Carol's 2nd Corolla",
      VIN: "ExampleVIN",
      bookingID: "002-EE037",
      journeyID: "0002-XXI7",
      journeyStartDateTime: "2019/04/06 08:22",
      journeyEndDateTime: "2019/04/09 08:47",
      odometerAtStart: 74000,
      odometerAtEnd: 74018, journeyStartLocation: "89080-789777",
      journeyEndLocation: "89080-789889",
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
  ],
  journeysHeadings: [
    {
      label: "Vehicle ID",
      field: "vehicleID",
    },
    {
      label: "VIN",
      field: "VIN",
    },
    {
      label: "Booking ID",
      field: "bookingID",
    },
    {
      label: "Journey ID",
      field: "journeyID",
    },
    {
      label: "Start Date-Time",
      field: "journeyStartDateTime",
    },
    {
      label: "End Date-Time",
      field: "journeyEndDateTime",
    },
    {
      label: "Odometer At Start (km)",
      field: "odometerAtStart",
    },
    {
      label: "Odometer At End (km)",
      field: "odometerAtEnd",
    },
    {
      label: "Starting Location",
      field: "journeyStartLocation",
    },
    {
      label: "Ending Location",
      field: "journeyEndLocation",
    },
    {
      label: "Date Created",
      field: "dateCreated",
    },
    {
      label: "Date Last Updated",
      field: "dateLastUpdated",
    },
  ],
  journeysDefaultRecord: {
    vehicleID: "Jimbo's Jeep",
    VIN: "ExampleVIN",
    bookingID: "055-EE037",
    journeyID: "0057-XXI7",
    journeyStartDateTime: "2019/04/06 08:22",
    journeyEndDateTime: "2019/04/09 08:47",
    odometerAtStart: 74000,
    odometerAtEnd: 74018,
    journeyStartLocation: "89080-789777",
    journeyEndLocation: "89080-789889",
    dateCreated: "2019/04/06",
    dateLastUpdated: "2019/04/06",
  },

  refuels: [
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      bookingID: "002-EE037",
      refuelID: "0001-XX00",
      fuelQuantity: 65.2,
      fuelPrice: 1.55,
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
    {
      vehicleID: "Jeff's Jeep",
      VIN: "ExampleVIN",
      bookingID: "001-JJI037",
      refuelID: "0001-XX00",
      fuelQuantity: 33.2,
      fuelPrice: 1.52,
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
    {
      vehicleID: "Carol's Corolla",
      VIN: "ExampleVIN",
      bookingID: "001-JJI037",
      refuelID: "0001-XX00",
      fuelQuantity: 33.2,
      fuelPrice: 1.52,
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
    {
      vehicleID: "Carol's 2nd Corolla",
      VIN: "ExampleVIN",
      bookingID: "001-JJI037",
      refuelID: "0001-XX00",
      fuelQuantity: 33.2,
      fuelPrice: 1.52,
      dateCreated: "2019/04/06",
      dateLastUpdated: "2019/04/06",
    },
  ],
  refuelsHeadings: [
    {
      label: "Vehicle ID",
      field: "vehicleID",
    },
    {
      label: "VIN",
      field: "VIN",
    },
    {
      label: "Booking ID",
      field: "bookingID",
    },
    {
      label: "Refuel ID",
      field: "refuelID",
    },
    {
      label: "Fuel Purchased (L)",
      field: "fuelQuantity",
    },
    {
      label: "Fuel Price ($/L)",
      field: "fuelPrice",
    },
    {
      label: "Date Created",
      field: "dateCreated",
    },
    {
      label: "Date Last Updated",
      field: "dateLastUpdated",
    },
  ],
  refuelsDefaultRecord: {
    vehicleID: "Jimbo's Jeep",
    VIN: "ExampleVIN",
    bookingID: "055-EE037",
    refuelID: "0191-XX00",
    fuelQuantity: 33.2,
    fuelPrice: 1.52,
    dateCreated: "2019/04/06",
    dateLastUpdated: "2019/04/06",
  },
};