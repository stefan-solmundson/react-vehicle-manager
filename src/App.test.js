// https://www.youtube.com/watch?v=3e1GHCA3GP0
// https://medium.com/@rossbulat/testing-in-react-with-jest-and-enzyme-an-introduction-99ce047dfcf8

import React from 'react';
import {render, cleanup} from '@testing-library/react';
import {MemoryRouter, Route} from 'react-router-dom';
// import {fakeAPI} from "../../setupTests";
// import {AddVehicle} from "./components/AddVehicle";
import '@testing-library/jest-dom/extend-expect'
// import {AppProvider} from "../../AppContext/AppContext";
import ReactDOM from 'react-dom';
import {App} from './App';
import renderer from "react-test-renderer";

import {localData as lData} from './LocalData';

import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';

Enzyme.configure({adapter: new Adapter()});

const tstate = lData;
let app = shallow(<App/>);
let appy = app.instance();

// beforeEach(() => {
//   // tstate = lData;
//   app = shallow(<App/>);
//   // appy.setState(tstate);
//   // console.log( "beforeEach" );
// });

// describe( 'Description of subtests', () => {
//   it( 'sums numbers', () => {
//     expect( 1 + 2 ).toEqual( 3 );
//     expect( 2 + 2 ).toEqual( 4 );
//   } );
// } );

// ---- State Modifying Tests

// addRecord
describe("checks addRecord() adds exactly 1 record.", () => {
  lData.pages.map((dataArrayName) => {
    it("adds the default record", () => {
      // console.log("appy.state[dataArrayName].length", appy.state[dataArrayName].length);
      let dataArrayLength1 = appy.state[dataArrayName].length;
      // expect(appy.state[dataArrayName]).toMatchSnapshot();
      appy.addRecord(
          appy.state[dataArrayName + "DefaultRecord"],
          (dataArrayName.substr(0, dataArrayName - 1) + "ID"),
          dataArrayName,
          null
      );
      // console.log("appy.state[dataArrayName].length", appy.state[dataArrayName].length);
      let dataArrayLength2 = appy.state[dataArrayName].length;
      expect(dataArrayLength2).toStrictEqual(dataArrayLength1 + 1);

      // WHY DOES REFRESHING THE APP ONLY WORK HERE INSIDE OF it() ???!!!
      // recreates the App instance "appy", THUS the constructor runs again the STATES are reset
      app = shallow(<App/>);
      appy = app.instance();
    });
  });
});

// deleteRecord
describe("checks deleteRecord() removes exactly 1 record.", () => {
  lData.pages.map((dataArrayName) => {
    it("deletes the first record", () => {
      // console.log("appy.state[dataArrayName].length", appy.state[dataArrayName].length);
      let dataArrayLength2 = appy.state[dataArrayName].length;
      appy.deleteRecord(
          appy.state[dataArrayName][0],
          (dataArrayName.substr(0, dataArrayName - 1) + "ID"),
          dataArrayName
      );
      // console.log("appy.state[dataArrayName].length", appy.state[dataArrayName].length);
      let dataArrayLength3 = appy.state[dataArrayName].length;
      expect(dataArrayLength3).toStrictEqual(dataArrayLength2 - 1);
      // console.log("appy.state[dataArrayName].length", appy.state[dataArrayName].length);

      // WHY DOES REFRESHING THE APP ONLY WORK HERE INSIDE OF it() ???!!!
      // recreates the App instance "appy", THUS the constructor runs again the STATES are reset
      app = shallow(<App/>);
      appy = app.instance();
    });
  });
});

// // editRecord
// describe("checks editRecord() can modify the record's vehicleID.", () => {
//   tstate.pages.map((dataArrayName) => {
//     it("changes the record's vehicleID to 'Test Vehicle ID'.", () => {
//       let _recordIdFieldName = dataArrayName.substr(0, dataArrayName.length - 1) + "ID";
//       let _record = appy.state[dataArrayName + "DefaultRecord"];
//       // expect(_record).toStrictEqual();
//       // vehicleID & _recordIdFieldName are REQUIRED, all other properties are made NULL
//       // vehicleID for updateOdometer(), which is inside of editRecord()
//       // _recordIdFieldName is require because this is a property in editRecord()
//       _record[_recordIdFieldName] = appy.state[dataArrayName][0][_recordIdFieldName];
//       _record.vehicleID = appy.state[dataArrayName][0].vehicleID;
//       Object.keys(_record).map(property => {
//         if (!(property === _recordIdFieldName || property === "vehicleID")) {
//           _record[property] = null;
//         }
//       });
//       // checks the first record of dataArrayName HAS NOT been changed yet
//       expect(appy.state[dataArrayName][0]).not.toStrictEqual(_record);
//       // console.log(_record);
//       // console.log(appy.state[dataArrayName][0]);
//       appy.editRecord(
//           _record,
//           _recordIdFieldName,
//           dataArrayName,
//           null
//       );
//       // checks the first record of dataArrayName HAS been changed yet
//       expect(appy.state[dataArrayName][0]).toStrictEqual(_record);
//
//       // DOESN'T WORK???
//       // WHY DOES REFRESHING THE APP ONLY WORK HERE INSIDE OF it() ???!!!
//       // recreates the App instance "appy", THUS the constructor runs again the STATES are reset
//       app = shallow(<App/>);
//       appy = app.instance();
//     });
//   });
// });

// sortArrayByField
describe("checks sortArrayByField correctly sorts.", () => {
  // tstate.contacts.map(contact => {
  it("sortArrayByField", () => {
    // console.log(appy.state.contacts);
    let failedOnce = false;
    let highestChar = "Z";
    appy.state.contacts.map((element, index) => {
      if (!(element.givenName[0] <= highestChar)) {
        failedOnce = true;
      }
      highestChar = element.givenName[0];
    });
    expect(failedOnce).toBeTruthy();
    //
    appy.sortArrayByField("contacts", "givenName", null);
    // appy.sortArrayByField("contacts", "familyName", null);
    //
    // console.log(appy.state.contacts);
    highestChar = "Z";
    appy.state.contacts.map((element, index) => {
      expect(element.givenName[0] <= highestChar).toBeTruthy();
      highestChar = element.givenName[0];
    });

    // DOESN'T WORK???
    // WHY DOES REFRESHING THE APP ONLY WORK HERE INSIDE OF it() ???!!!
    // recreates the App instance "appy", THUS the constructor runs again the STATES are reset
    app = shallow(<App/>);
    appy = app.instance();
  });
});

// searchArray
describe("checks searchArray() can find a contact.", () => {
  // tstate.contacts.map(contact => {
  it("searchArray", () => {
    // console.log(appy.state.contacts);
    expect(appy.state.contacts[0].givenName).not.toStrictEqual("Amy");
    appy.searchArray("contacts", "Amy");
    // console.log(appy.state.contacts);
    expect(appy.state.contacts[0].givenName).toStrictEqual("Amy");

    // DOESN'T WORK???
    // WHY DOES REFRESHING THE APP ONLY WORK HERE INSIDE OF it() ???!!!
    // recreates the App instance "appy", THUS the constructor runs again the STATES are reset
    app = shallow(<App/>);
    appy = app.instance();
  });
});

// ---- Some Notes

// // https://www.youtube.com/watch?v=3e1GHCA3GP0
// // https://medium.com/@rossbulat/testing-in-react-with-jest-and-enzyme-an-introduction-99ce047dfcf8
//
//
// import React from 'react';
// import {render} from '@testing-library/react';
// import {MemoryRouter, Route} from 'react-router-dom';
// // import {fakeAPI} from "../../setupTests";
// // import {AddVehicle} from "./AddVehicle";
// import '@testing-library/jest-dom/extend-expect'
// // import {AppProvider} from "../../AppContext/AppContext";
//
//
// import ReactDOM from 'react-dom';
// import App from './App';
//
// import {render, cleanup } from "@testing-library/react";
// import "jest-dom/extend-expect";
//
// import renderer from "react=test=rendered";
//
// beforeEach(() => {
//   initialState = {};
//
//   tree = (
//       <MemoryRouter initialEntries={[`/`]}>
//         <App/>
//       </MemoryRouter>
//   )
// })
//
//   it('renders without crashing', () => {
//     const div = document.createElement('div');
//     ReactDOM.render(<App/>, div);
//     ReactDOM.unmountComponentAtNode(div);
//   });
//   //
//   // afterEach(cleanup);
//   //
//   // it("renders thing correctly", () => {
//   //   const {getByTestId} = render(<Button label="save"/>);
//   //   expect(getByTestId("button")).toHaveTextContent("save");
//   // });
//   //
//   // it("matches snapshot", () => {
//   //   renderer.create(<Button label="save"/>).toJSON();
//   //   expect(tree).toMatchSnapShot();
//   // });
//
// // ----
// // export const testTests = () => {
// //
// //   return(<>
// //   describe('Description of subtests', () => {
// //     it('sums numbers', () => {
// //       expect(1 + 2).toEqual(3);
// //       expect(2 + 2).toEqual(4);
// //     });
// //   })
// //     </>
// // )
// // };
