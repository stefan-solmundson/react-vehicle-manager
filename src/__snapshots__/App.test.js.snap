// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks sortArrayByField correctly sorts. changes the record's vehicleID to 'Test Vehicle ID'. 1`] = `
Array [
  Object {
    "company": "",
    "contactID": "1",
    "contactType": "Friend",
    "email": "steve.johnson@gmail.com",
    "familyName": "Johnson",
    "givenName": "Steve",
    "homePhone": "",
    "mobile": "1111222233",
  },
  Object {
    "company": "Hilary's Fish 'n' Chips",
    "contactID": "2",
    "contactType": "Work",
    "email": "hilary.kirk@gmail.com",
    "familyName": "Kirk",
    "givenName": "Hilary",
    "homePhone": "",
    "mobile": "2222333344",
  },
  Object {
    "company": "JeffLifts",
    "contactID": "3",
    "contactType": "Sport",
    "email": "jeff.yota@gmail.com",
    "familyName": "Yota",
    "givenName": "Jeffery",
    "homePhone": "",
    "mobile": "3333444455",
  },
  Object {
    "company": "OfficeWorks",
    "contactID": "31",
    "contactType": "Work",
    "email": "amy.bjorne@gmail.com",
    "familyName": "Bjorne",
    "givenName": "Amy",
    "homePhone": "44448888",
    "mobile": "0000111122",
  },
]
`;

exports[`checks sortArrayByField correctly sorts. changes the record's vehicleID to 'Test Vehicle ID'. 2`] = `
Array [
  Object {
    "company": "",
    "contactID": "1",
    "contactType": "Friend",
    "email": "steve.johnson@gmail.com",
    "familyName": "Johnson",
    "givenName": "Steve",
    "homePhone": "",
    "mobile": "1111222233",
  },
  Object {
    "company": "JeffLifts",
    "contactID": "3",
    "contactType": "Sport",
    "email": "jeff.yota@gmail.com",
    "familyName": "Yota",
    "givenName": "Jeffery",
    "homePhone": "",
    "mobile": "3333444455",
  },
  Object {
    "company": "Hilary's Fish 'n' Chips",
    "contactID": "2",
    "contactType": "Work",
    "email": "hilary.kirk@gmail.com",
    "familyName": "Kirk",
    "givenName": "Hilary",
    "homePhone": "",
    "mobile": "2222333344",
  },
  Object {
    "company": "OfficeWorks",
    "contactID": "31",
    "contactType": "Work",
    "email": "amy.bjorne@gmail.com",
    "familyName": "Bjorne",
    "givenName": "Amy",
    "homePhone": "44448888",
    "mobile": "0000111122",
  },
]
`;
